### ISSUE TITLE
IDE0016 provides invalid codefix on generic type without constraint

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/roslyn#22926]
**Version Used**: 
Visual Studio 15.4.1

**Steps to Reproduce**:
```csharp
class A<T>
{
    T x;
    public A(T t)
    {
        if (t == null) throw new ArgumentNullException();
        x = t;
    }
}
```

**Expected Behavior**:
T is not constrained, so `??` cannot be applied on it.

**Actual Behavior**:
Code fix to
```csharp
x = t ?? throw new ArgumentNullException();
```

How about allowing `??` on unconstrained generic type? For value type, just ignore it, as it implies `== null`.

**Comment by sharwell** [comment]
This appears to be a duplicate of #17754.

**Comment by sharwell** [comment]
Reopening since this should have been fixed in 15.3, but @huoyaoyuan was observing it in 15.4.1.
